plugin/service/host/plugin.go

```diff
@@ -673,12 +677,19 @@ func awsInstanceToHost(instance types.Instance) (*pb.ListHostsResponseHost, erro
 // credentials, the host listing functionality as well as the filters, if any
 // are passed in. This function can therefore be used for both host catalog and
 // host set validation.
-func dryRunValidation(ctx context.Context, state *awsCatalogPersistedState, filters ...types.Filter) *status.Status {
+func dryRunValidation(ctx context.Context, state *awsCatalogPersistedState, catalogAttributes *CatalogAttributes, filters ...types.Filter) *status.Status {
```

Comment:
~if we only need the `ec2Option`, why don't we just pass in `ec2Option` as a parameter instead of all the `catalogAttributes` where we need to do validation and also build ec2 options in `dryRunValidation`

Reply:
I applied this change